### PR TITLE
CASMTRIAGE-6548 add the execution of upload-rebuild-templates.sh before running IUF

### PR DESCRIPTION
**Pull request by leliasen-hpe**
# Description

This changes adds the execution of upload-rebuild-templates.sh before running IUF steps. This updates CSM product versions and makes sure the IUF workflowTemplates used by Argo are updated.

This is necessary because we ran into a specific case where IUF was trying to install the incorrect product version. We believe this happened because of an ordering issue, where a newer version of docs-csm rpm is installed after a CSM upgrade. If a newer version of docs-csm is installed but this script is not run, then IUF might be referencing old versions. This script is run in prerequisistes.sh which is likely why we don't run into this problem more often.

# Checklist

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: operations/iuf/workflows/product_delivery.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ This section ensures the product content is loaded onto the system and available
 section of the _HPE Cray EX System Software Stack Installation and Upgrade Guide for CSM (S-8052)_ provides a table that summarizes which product documents contain information or actions for the `process-media` or `pre-install-check` stages.
 Refer to that table and any corresponding product documents before continuing to the next step.
 
+1. Run `upload-rebuild-tempaltes.sh` to update all the workflows that will be used by IUF and to ensure the correct CSM product versions will be used by IUF.
```
Comment (don-bahls-hpe) [inline operations/iuf/workflows/product_delivery.md]: Typo in "templates"

File: operations/iuf/workflows/product_delivery.md [hunk 2]
```diff
@@ -14,6 +14,14 @@ This section ensures the product content is loaded onto the system and available
...
+
+    (`ncn-m001#`) Execute the `upload-rebuild-tempaltes.sh` script.
```
Comment (don-bahls-hpe) [inline operations/iuf/workflows/product_delivery.md]: Same here

**Review by don-bahls-hpe — APPROVED** [review]

File: operations/iuf/workflows/product_delivery.md (hunk 1 above)
Comment (leliasen-hpe) [reply operations/iuf/workflows/product_delivery.md]: fixed

File: operations/iuf/workflows/product_delivery.md (hunk 2 above)
Comment (leliasen-hpe) [reply operations/iuf/workflows/product_delivery.md]: fixed, thank you

**Review by rustydb — APPROVED** [review]

**Review by nrockershousen — APPROVED** [review]

**Pull request merged**